File: solutions/csharp/phone-number/1/PhoneNumber.cs
using System;
using System.Linq;

public class PhoneNumber
{
    //private static Regex digits = new Regex(@"\d");
    public static string Clean(string phoneNumber)
    {
        var digits = string.Concat(phoneNumber.Where(char.IsDigit));
        if (digits.Length < 10) throw new ArgumentException("too short");        
        if (digits.Length == 11) {
          if (digits[0] != '1')  throw new ArgumentException("country code needs to be 1");
          digits = digits.Substring(1);
        } 
        if (digits.Length > 11) throw new ArgumentException("too long");
        if (digits[0] == '0' || digits[0] == '1') throw new ArgumentException("invalid area code");
        if (digits[3] == '0' || digits[3] == '1') throw new ArgumentException("invalid exchange code");
        return digits;
    }
}

File: solutions/kotlin/phone-number/1/src/main/kotlin/PhoneNumber.kt
class PhoneNumber {
    val number: String

    constructor(number: String) {
        var digits = number.filter { it.isDigit() }

        when (digits.length) {
            10 -> {}
            11 -> {
                if (digits[0] != '1') {
                    throw IllegalArgumentException()
                } else {
                    digits = digits.drop(1)
                }
            }
            else -> throw IllegalArgumentException()
        }

        require (digits[0] !in "01")
        require (digits[3] !in "01")

        this.number = digits
    }
}
